RecoCTPPS/PixelLocal/src/RPixRoadFinder.cc

```diff
@@ -0,0 +1,128 @@
+
+#include "RecoCTPPS/PixelLocal/interface/RPixRoadFinder.h"
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+
+//needed for the geometry:
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "TMatrixD.h"
+
+#include <vector>
+#include <memory>
+#include <string>
+#include <iostream>
+
+//------------------------------------------------------------------------------------------------//
+
+RPixRoadFinder::RPixRoadFinder(edm::ParameterSet const& parameterSet) : 
+	RPixDetPatternFinder(parameterSet),
+	param_(parameterSet){
+
+	verbosity_ = param_.getUntrackedParameter<int> ("RPixVerbosity");
+  roadRadius_ = param_.getParameter<double>("RPixRoadRadius");
+  minRoadSize_ = param_.getParameter<int>("RPixMinRoadSize");
+  maxRoadSize_ = param_.getParameter<int>("RPixMaxRoadSize");
+
+}
+
+//------------------------------------------------------------------------------------------------//
+
+RPixRoadFinder::~RPixRoadFinder(){
+}
+
+//------------------------------------------------------------------------------------------------//
+
+void RPixRoadFinder::findPattern(){
+
+  Road temp_all_hits;
+  temp_all_hits.clear();
+
+// convert local hit sto global and push them to a vector
+  for(const auto & ds_rh2 : hitVector_){
+    uint32_t myid = ds_rh2.id;
+    for (const auto & _rh : ds_rh2.data){
```

Comment:
please do not use leading underscores [even trailing underscore is only for data members]